ARCHITECTURE.md

```diff
@@ -0,0 +1,67 @@
+# Salesforce CLI Architecture
+
+## Plugin Architecture
+
+The Salesforce CLI is built using [oclif](oclif.io), which is an open source CLI framework owned and maintained by Salesforce. The primary benefit of using oclif is that it allows us to develop CLI plugins for specific feature areas. Some plugins are bundled with every CLI installation, while others can be installed if and when a user chooses.
+
+This architecture allows us to empower internal teams and external contributors to own and maintain their CLI commands, without having to create an entirely new CLI.
```

Comment:
there's also some separation of concerns benefits.  

you don't have to worry about all the other parts, just yours
keep changes isolated and flowing ("hold on...ours isn't ready to release!")
allow customers to preview features/fixes by manually installing the updated plugin before it ships in the CLI